CONTRIBUTING.md

```diff
@@ -301,6 +290,52 @@ git add index.rst jit.rst
 ...
 ```
 
+#### Building C++ Documentation
+For C++ documentation (https://pytorch.org/cppdocs), we use
+[Doxygen](http://www.doxygen.nl/) and then convert it to
+[Sphinx](http://www.sphinx-doc.org/) via
+[Breathe](https://github.com/michaeljones/breathe) and
+[Exhale](https://github.com/svenevs/exhale). Check the [Doxygen
+reference](http://www.stack.nl/~dimitri/doxygen/manual/index.html) for more
+information on the documentation syntax.
+
+We run Doxygen in CI (Travis) to verify that you do not use invalid Doxygen
+commands. To run this check locally, run `./check-doxygen.sh` from inside
+`docs/cpp`.
+
+To build the documentation, follow the same steps as above, but run them from
+`docs/cpp` instead of `docs`.
+
+### Previewing changes
+
+It is helpful when submitting a PR that changes the docs to provide a rendered
+version of the result. You can do this by using GitHub pages to host the docs
+you have built. To do so, follow [these steps](https://guides.github.com/features/pages/)
+to make a repo to host your changed documentation.
+
+GitHub pages expects to be hosting a Jekyll generated website which does not work
+well with the static resource paths used in the PyTorch documentation. To get around
+this, you must add an empty file called `.nojekyll` to your repo.
+
+```bash
+cd your_github_pages_repo
+touch .nojekyll
+git add .
+git commit
+git push
+```
+
+Then, copy built documentation and push the changes:
+
+```bash
+cd your_github_pages_repo
+cp -r ~/my_pytorch_path/docs/build/html/* .
```

Comment:
I've always previewed documentation by just opening files from this directory. Is that something we could recommend as well?

Reply:
Do you mean opening it on GitHub with GitHub's rst rendering (e.g. [this page](https://github.com/pytorch/pytorch/blob/master/docs/source/jit.rst))? I've found that that doesn't always match the website since most directives aren't followed (e.g. `automodule`/`autofunction`)

Reply:
No, I mean opening `~/my_pytorch_path/docs/build/html/index.html` in my local browser

Reply:
This is more for reviewing so other people can see it, I think the section titles were a little confusing so I've split them up

Reply:
from offline: the instructions recommended starting an http server but that's not necessary since you can open the files in a browser, so the latest commit removes that and recommends just opening the files